### PR TITLE
Monitoring Dashboards: Add variable dropdowns

### PR DESCRIPTION
**Pull request by kyoto**
Use the dashboard JSON definition's `templating` sections to generate
dropdowns for the dashboard variables.

This PR does not handle the case where a variable's `query` contains the another variable. That will be addressed with a separate PR.

### CODE REVIEW

File: frontend/public/components/monitoring/dashboards/index.tsx [hunk 1]
```diff
@@ -172,13 +233,21 @@ const Board: React.FC<{ board: string; pollInterval: number; timespan: number }>
...
-  const [pollInterval, setPollInterval] = React.useState(
+const MonitoringDashboardsPage_: React.FC<MonitoringDashboardsPageProps> = ({
+  clearVariables,
+  patchVariable,
+}) => {
+  const [pollInterval, setPollInterval] = React.useState<number>(
     parsePrometheusDuration(defaultPollInterval),
   );
-  const [timespan, setTimespan] = React.useState(parsePrometheusDuration(defaultTimespan));
+  const [timespan, setTimespan] = React.useState<number>(parsePrometheusDuration(defaultTimespan));
```
Comment (spadgett) [inline frontend/public/components/monitoring/dashboards/index.tsx]: If `parsePrometheusDuration` has a return type, the type should be inferred

Comment (kyoto) [reply frontend/public/components/monitoring/dashboards/index.tsx]: Thanks, removed by #3965

**Pull request merged**